activerecord/lib/active_record/associations/singular_association.rb

```diff
@@ -3,7 +3,7 @@ module Associations
     class SingularAssociation < Association #:nodoc:
       # Implements the reader method, e.g. foo.bar for Foo.has_one :bar
       def reader(force_reload = false)
-        if force_reload
+        if force_reload && klass
```

Comment:
I think the problem now, is that instead of getting the error, the `force_reload` will not work for `polymorphic`. as it will not call `reload`.